File: leetcode.com/python/1_Two_Sum.py
# https://www.algoexpert.io/questions/Two%20Number%20Sum

class Solution(object):
    def twoSum(self, nums, target):
        indices = []
        numIdxDict = {}
        for idx, num in enumerate(nums):
            numIdxDict[num] = idx
        for idx, num in enumerate(nums):
            expectedNum = target - num
            if expectedNum in numIdxDict:
                if idx != numIdxDict[expectedNum]:
                    indices.extend([idx, numIdxDict[expectedNum]])
                    break
        return sorted(indices)


sol = Solution()
# nums = [2, 7, 11, 15]
# target = 9
nums = [3,2,4]
target = 6
out = sol.twoSum(nums, target)
print("Res: ", out)

File: leetcode.com/swift/DS_ALGO_PRAC.playground/Sources/swift/1_Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numIdxMap = [Int:Int]()
        for i in 0..<nums.count {
            numIdxMap[nums[i]] = i
        }
        for i in 0..<nums.count {
            let num = nums[i]
            if let j = numIdxMap[target - num], j != i {
                return [i, j]
            }
        }
        return [-1, -1]
    }
}
